polkadot/xcm/docs/src/cookbook/other_parachain_tokens/parachain/xcm_config.rs

```diff
@@ -0,0 +1,214 @@
+// Copyright Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # XCM Configuration
+
+use frame::{
+	deps::frame_system,
+	runtime::prelude::*,
+	traits::{Everything, Nothing},
+};
+use xcm::v4::prelude::*;
+use xcm_builder::{
+	AccountId32Aliases, DescribeAllTerminal, DescribeFamily, EnsureXcmOrigin,
+	FrameTransactionalProcessor, HashedDescription, IsConcrete,
+	SignedToAccountId32,
+};
+use xcm_executor::XcmExecutor;
+
+use super::{AccountId, Balances, MessageQueue, Runtime, RuntimeCall, RuntimeEvent, RuntimeOrigin};
+
+parameter_types! {
+	pub RelayLocation: Location = Location::parent();
+	pub ThisNetwork: NetworkId = NetworkId::Polkadot;
+}
+
+pub type LocationToAccountId = (
+	HashedDescription<AccountId, DescribeFamily<DescribeAllTerminal>>,
+	AccountId32Aliases<ThisNetwork, AccountId>,
+);
+
+/// Configuration related to asset transactors
+#[docify::export]
+mod asset_transactor {
+	use super::*;
+
+	use frame::traits::EverythingBut;
+	use xcm_builder::{
+		FungibleAdapter, FungiblesAdapter, IsConcrete, MatchedConvertedConcreteId, MintLocation,
+		NoChecking, StartsWith,
+	};
+	use xcm_executor::traits::Identity;
+
+	use super::super::{
+		AccountId, Balance, Balances, ForeignAssets, PolkadotXcm,
+	};
+
+	parameter_types! {
+		pub LocalPrefix: Location = Location::here();
+		pub CheckingAccount: AccountId = PolkadotXcm::check_account();
+	}
+
+	/// AssetTransactor for handling the chain's native token.
+	pub type FungibleTransactor = FungibleAdapter<
+		// What implementation of the `fungible::*` traits do we want to use?
+		Balances,
+		// What tokens should be handled by this transactor?
+		IsConcrete<LocalPrefix>,
+		// How do we convert an XCM Location into a local account id?
+		LocationToAccountId,
+		// The type for account ids, only needed because `fungible` is generic over it.
+		AccountId,
+		// Tracking teleports.
+		(),
+	>;
+
+	/// Type that matches foreign assets.
+	/// We do this by matching on all possible Locations and excluding the ones
+	/// inside our local chain.
+	pub type ForeignAssetsMatcher = MatchedConvertedConcreteId<
+		Location, // Asset id.
+		Balance, // Balance type.
+		EverythingBut<StartsWith<LocalPrefix>>, // Location matcher.
+		Identity, // How to convert from Location to AssetId.
```

Comment:
This is only possible because `type AssetId = xcm::v4::Location;` above, right?

If I mis-configure the two, will it be detected by the compiler?

Reply:
Correct, I can add a remark on the subject. I'm also thinking the docify sections should be smaller so I can talk about each one.